packages/client/hmi-client/src/components/widgets/tera-output-dropdown.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+	<Dropdown
+		v-if="options"
+		class="output-dropdown"
+		placeholder="Select an output"
+		:model-value="output"
+		:options="options"
+		option-value="id"
+		option-label="label"
+		option-group-children="items"
+		option-group-label="label"
+		@update:model-value="emit('update:output', $event)"
+		:loading="isLoading"
+	>
+		<template #optiongroup="slotProps">
+			<span class="dropdown-option-group">{{ slotProps.option?.label }}</span>
+		</template>
+		<template #option="slotProps">
+			<div class="dropdown-option">
+				<Checkbox
+					v-if="isSelectable"
+					@click.stop
+					:model-value="slotProps.option?.isSelected"
+					@update:model-value="emit('update:selection', slotProps.option?.id)"
+					binary
+				/>
+				<span>{{ slotProps.option?.label }}</span>
+				<span
+					v-if="slotProps.option?.status === WorkflowPortStatus.CONNECTED"
+					class="connection-indicator"
+					><i class="pi pi-link" />Connected</span
+				>
+			</div>
+		</template>
+	</Dropdown>
+</template>
+
+<script setup lang="ts">
+import { WorkflowOutput, WorkflowPortStatus } from '@/types/workflow';
+import Dropdown from 'primevue/dropdown';
+import Checkbox from 'primevue/checkbox';
+
+defineProps<{
+	options?: WorkflowOutput<any>[] | { label: string; items: WorkflowOutput<any>[] }[];
+	output?: WorkflowOutput<any>['id'];
+	isSelectable?: boolean;
+	isLoading?: boolean;
+}>();
```

Comment:
Passing props like this should help https://vuejs.org/guide/components/props.html#prop-validation

Reply:
makes sense, Ive changed the options and output props so that they are no longer optional in this component:
```
defineProps<{
	options: WorkflowOutput<any>[] | { label: string; items: WorkflowOutput<any>[] }[];
	output: WorkflowOutput<any>['id'];
	isSelectable?: boolean;
	isLoading?: boolean;
}>();
```